lightningd/peer_control.c

```diff
@@ -2091,6 +2091,8 @@ static bool peer_start_channeld(struct peer *peer,
 	if (peer->ld->config.ignore_fee_limits)
 		log_debug(peer->log, "Ignoring fee limits!");
 
+	peer->direction = pubkey_cmp(&peer->ld->id, &peer->id) > 0;
```

Comment:
Hoist the channel_direction() macro out of gossipd perhaps?

Reply:
It's in `routing.h`, so just including that should suffice.